src/main/java/org/candlepin/subscriptions/tally/TallyTaskFactory.java

```diff
@@ -52,6 +54,18 @@ public Task build(TaskDescriptor taskDescriptor) {
         if (taskDescriptor.getTaskType() == TaskType.UPDATE_SNAPSHOTS) {
             return new UpdateAccountSnapshotsTask(snapshotController, taskDescriptor.getArg("accounts"));
         }
+
+        if (taskDescriptor.getTaskType() == TaskType.UPDATE_HOURLY_SNAPSHOTS) {
+            String accountNumber = taskDescriptor.getArg("accountNumber").get(0);
+            String startDateTime = taskDescriptor.getArg("startDateTime").get(0);
+            String endDateTime = taskDescriptor.getArg("endDateTime").get(0);
```

Comment:
Agreed, I kind of twitched when I wrote these lines...but where we've been up in the air about where we're going to this task framework, I didn't want to get too deep in the mud of how to do this elegantly.

At the very least, I'll for sure null check these to prevent the NPE and throw a more useful exception.  Open to anymore recommendations for this as well.

Reply:
:+1: